File: practica_5/algebra/poo/pro_algebra.py
import math

class ProAlgebra:
    def __init__(self, a=1, b=2, c=1):
        self.a = a
        self.b = b
        self.c = c

    def set_a(self, a): self.a = a
    def set_b(self, b): self.b = b
    def set_c(self, c): self.c = c

    def get_discriminante(self):
        return self.b**2 - 4 * self.a * self.c

    def get_raiz1(self):
        if self.get_discriminante() < 0:
            return float('nan')
        return (-self.b + math.sqrt(self.get_discriminante())) / (2 * self.a)

    def get_raiz2(self):
        if self.get_discriminante() < 0:
            return float('nan')
        return (-self.b - math.sqrt(self.get_discriminante())) / (2 * self.a)

    def __str__(self):
        discriminante = self.get_discriminante()
        if discriminante < 0:
            return "La ecuación no tiene raíces reales."
        elif discriminante == 0:
            return f"La ecuación tiene una raíz real: {self.get_raiz1()}"
        else:
            return f"La ecuación tiene dos raíces reales: {self.get_raiz1()} y {self.get_raiz2()}"


e1 = ProAlgebra()
print(e1)

File: practica_5/algebra/poo/pro_algebra.java
package practica_5.algebra.poo;
public class pro_algebra {
    private double a, b, c;
    public pro_algebra() {
        this.a = 1;
        this.b = 2;
        this.c = 1;}
    public pro_algebra(double a, double b, double c) {
        this.a = a;
        this.b = b;
        this.c = c;
    }
    public void setA(double a) { this.a = a; }
    public void setB(double b) { this.b = b; }
    public void setC(double c) { this.c = c; }
    public double getDiscriminante() {
        return b * b - 4 * a * c;}
    public double getRaiz1() {
        if (getDiscriminante() < 0) {
            return Double.NaN;}
        return (-b + Math.sqrt(getDiscriminante())) / (2 * a);
    }
    public double getRaiz2() {
        if (getDiscriminante() < 0) {
            return Double.NaN;
        }
        return (-b - Math.sqrt(getDiscriminante())) / (2 * a);
    }
    public String toString() {
        double discriminante = getDiscriminante();
        if (discriminante < 0) {
            return "La ecuación no tiene raíces reales.";
        } else if (discriminante == 0) {
            return "La ecuación tiene una raíz real: " + getRaiz1();
        } else {
            return "La ecuación tiene dos raíces reales: " + getRaiz1() + " y " + getRaiz2();
        }
    }
    public static void main(String[] args) {
        pro_algebra e1 = new pro_algebra();
        System.out.println(e1.toString());
    }
}
